### ISSUE TITLE
Magento 2 version 2.4.7-p2 “Page Builder was rendering for 5 seconds without releasing locks” error.

### ISSUE THREAD

**Issue by sunit9** [magento/magento2#39076]
### Preconditions and environment

Magento 2.4.7-p2

### Steps to reproduce

1) Login to magento admin
2) Content -> Block -> Add new block
3) Add Block Title
4) Add Identifier
5) Click on Save

### Expected result

Block should be save successfully.

### Actual result

On screen only loader display and in console getting error.
Page Builder was rendering for 5 seconds without releasing locks
![2024-08-21_10-22](https://github.com/user-attachments/assets/5d21560d-2120-4123-b1cb-023c45853316)

### Triage and priority

- [ ] Severity: **S0** _- Affects critical data or functionality and leaves users without workaround._
- [X] Severity: **S1** _- Affects critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S2** _- Affects non-critical data or functionality and forces users to employ a workaround._
- [ ] Severity: **S3** _- Affects non-critical data or functionality and does not force users to employ a workaround._
- [X] Severity: **S4** _- Affects aesthetics, professional look and feel, “quality” or “usability”._

**Comment by Bashev** [comment]
looks like same with this: https://github.com/magento/magento2/issues/37516 and https://github.com/magento/magento2/issues/37798

**Comment by sunit9** [comment]
I have check in firefox but not working.
My chrome version is Version 127.0.6533.119 (Official Build) (64-bit)
I check same things for 2.4.6-p2 and it's working fine.

**Comment by medigeek** [comment]
I think your problem is with mageno 2.4 and CORS - apache htaccess or nginx requires some tweaking

Check your browser's javascript console log.

Access to script at 'https://example.com/static/versionXXXXXXXXXX/adminhtml/Magento/backend/en_US/requirejs-config.js' from origin 'null' has been blocked by CORS policy: No 'Access-Control-Allow-Origin' header is present on the requested resource.

The solution is explained here:
https://magento.stackexchange.com/a/170726

Had the same issue. The above worked for a support case. Hopefully it will work for you too.

**Comment by engcom-Bravo** [comment]
@magento give me 2.4.7-p2 instance

**Comment by engcom-Bravo** [comment]
@medigeek Thanks for your Contribution!!

Hi @sunit9,

Thanks for your reporting and collaboration.

Could you please refer this comment https://github.com/magento/magento2/issues/39076#issuecomment-2303063100. which resolves your issue and please let us know if you are still facing any issue.

Thanks.

**Comment by sunit9** [comment]
I have change in following file
vendor/magento/module-page-builder/view/adminhtml/web/template/page-builder.html

`<iframe attr="id: 'render_frame_' + id" sandbox="allow-scripts" style="position: absolute; width:0; height:0; border: none;"></iframe>`

To

`<iframe attr="id: 'render_frame_' + id" sandbox="allow-scripts allow-same-origin" style="position: absolute; width:0; height:0; border: none;"></iframe>`

In sandbox i append "**allow-same-origin**"

It works for me. I do not have idea it's good solution or not.

**Comment by chrisoliver90** [comment]
Adobe Commerce offered us the following patch that fixed this:

`diff --git a/vendor/magento/module-backend/ViewModel/RequireJsConfigModifierInterface.php b/vendor/magento/module-backend/ViewModel/RequireJsConfigModifierInterface.php
new file mode 100644
index 000000000000..07bd7ca989e6
--- /dev/null
+++ b/vendor/magento/module-backend/ViewModel/RequireJsConfigModifierInterface.php
@@ -0,0 +1,22 @@
+<?php
+/**
+ * Copyright 2024 Adobe
+ * All Rights Reserved.
+ */
+declare(strict_types=1);
+
+namespace Magento\Backend\ViewModel;
+
+/**
+ * View model interface for requirejs configuration modifier
+ */
+interface RequireJsConfigModifierInterface
+{
+    /**
+     * Modifies requirejs configuration
+     *
+     * @param array $config requirejs configuration
+     * @return array
+     */
+    public function modify(array $config): array;
+}
diff --git a/vendor/magento/module-backend/view/adminhtml/templates/page/js/require_js.phtml b/vendor/magento/module-backend/view/adminhtml/templates/page/js/require_js.phtml
index 6fa41e107995..1b1a5d7fcaf3 100644
--- a/vendor/magento/module-backend/view/adminhtml/templates/page/js/require_js.phtml
+++ b/vendor/magento/module-backend/view/adminhtml/templates/page/js/require_js.phtml
@@ -5,12 +5,20 @@
  */
 
 /** @var \Magento\Framework\View\Helper\SecureHtmlRenderer $secureRenderer */
+/** @var \Magento\Backend\Block\Page\RequireJs $block */
+
+$requireJsConfig = [
+    'baseUrl' => $block->getViewFileUrl('/'),
+];
+
+$configModifier = $block->getConfigModifier();
+$requireJsConfig = $configModifier instanceof \Magento\Backend\ViewModel\RequireJsConfigModifierInterface
+    ? $configModifier->modify($requireJsConfig)
+    : $requireJsConfig;
 
 $scriptString = '
     var BASE_URL = \'' . /* @noEscape */ $block->getUrl('*') . '\';
     var FORM_KEY = \'' . /* @noEscape */ $block->getFormKey() . '\';
-    var require = {
-        \'baseUrl\': \'' . /* @noEscape */ $block->getViewFileUrl('/') . '\'
-    };';
-
-echo /* @noEscape */ $secureRenderer->renderTag('script', [], $scriptString, false);
+    var require = ' . /* @noEscape */ json_encode($requireJsConfig) .';';
+?>
+<?= /* @noEscape */ $secureRenderer->renderTag('script', [], $scriptString, false) ?>
diff --git a/vendor/magento/module-page-builder/ViewModel/StageRenderFrameRequireJsConfigModifier.php b/vendor/magento/module-page-builder/ViewModel/StageRenderFrameRequireJsConfigModifier.php
new file mode 100644
index 000000000..1ae244b03
--- /dev/null
+++ b/vendor/magento/module-page-builder/ViewModel/StageRenderFrameRequireJsConfigModifier.php
@@ -0,0 +1,37 @@
+<?php
+/**
+ * Copyright 2024 Adobe
+ * All Rights Reserved.
+ */
+declare(strict_types=1);
+
+namespace Magento\PageBuilder\ViewModel;
+
+use Magento\Backend\ViewModel\RequireJsConfigModifierInterface;
+use Magento\Framework\View\Element\Block\ArgumentInterface;
+
+/**
+ * Modifies requirejs configuration for the stage render frame
+ *
+ * Override the text! plugin within the iframe to ensure we can pipe any XHR requests through to the parent window
+ * as the same origin policy will not allow us to load the templates within this iframe.
+ * It is important that this mapping is configured before requirejs-config.js to ensure the text! plugin is overridden
+ * for all requests.
+ */
+class StageRenderFrameRequireJsConfigModifier implements ArgumentInterface, RequireJsConfigModifierInterface
+{
+    /**
+     * @inheritDoc
+     */
+    public function modify(array $config): array
+    {
+        $config['map']['*'] = array_merge(
+            $config['map']['*'] ?? [],
+            [
+                'text' => 'Magento_PageBuilder/js/master-format/render/requirejs/text',
+                'Magento_PageBuilder/js/events' => 'Magento_PageBuilder/js/master-format/render/events'
+            ]
+        );
+        return $config;
+    }
+}
diff --git a/vendor/magento/module-page-builder/view/adminhtml/layout/pagebuilder_stage_render.xml b/vendor/magento/module-page-builder/view/adminhtml/layout/pagebuilder_stage_render.xml
index 6a95e8e87..c40005870 100644
--- a/vendor/magento/module-page-builder/view/adminhtml/layout/pagebuilder_stage_render.xml
+++ b/vendor/magento/module-page-builder/view/adminhtml/layout/pagebuilder_stage_render.xml
@@ -14,6 +14,13 @@
         <remove src="css/styles.css"/>
     </head>
     <body>
+        <referenceBlock name="require.js">
+            <arguments>
+                <argument name="config_modifier" xsi:type="object">
+                    Magento\PageBuilder\ViewModel\StageRenderFrameRequireJsConfigModifier
+                </argument>
+            </arguments>
+        </referenceBlock>
         <referenceContainer name="backend.page" remove="true"/>
         <referenceContainer name="menu.wrapper" remove="true"/>
         <referenceContainer name="root">
diff --git a/vendor/magento/module-page-builder/view/adminhtml/templates/stage/render.phtml b/vendor/magento/module-page-builder/view/adminhtml/templates/stage/render.phtml
index 6e96a2757..2586df2b3 100644
--- a/vendor/magento/module-page-builder/view/adminhtml/templates/stage/render.phtml
+++ b/vendor/magento/module-page-builder/view/adminhtml/templates/stage/render.phtml
@@ -10,33 +10,16 @@
  */
 ?>
 
-<?php
-/**
- * Override the text! plugin within the iframe to ensure we can pipe any XHR requests through to the parent window
- * as the same origin policy will not allow us to load the templates within this iframe.
- */
-?>
 <?php
 $pageBuilderConfig = $block->getPageBuilderConfig();
 
-$script = <<<SCRIPT
-   require.config({
-       'map': {
-           '*': {
-               'text': 'Magento_PageBuilder/js/master-format/render/requirejs/text',
-               'Magento_PageBuilder/js/events': 'Magento_PageBuilder/js/master-format/render/events'
-           }
-       }
-   });
-SCRIPT;
-
 /**
  * To be able to override the text plugin we need the Magento template engine to be used, as the template engine
  * within lib has a dependency on the text! plugin we need to ensure we set the template engine before the
  * dependency blocks us. If we try to just override using the RequireJS config above our !text plugin will never
  * get overridden as our template engine cannot load.
  */
-$script .= <<<SCRIPT
+$script = <<<SCRIPT
     require([
         'ko',
         'Magento_Ui/js/lib/knockout/template/engine'
diff --git a/vendor/magento/module-page-builder/view/adminhtml/web/template/page-builder.html b/vendor/magento/module-page-builder/view/adminhtml/web/template/page-builder.html
index 8b495ce72..1ff74d860 100644
--- a/vendor/magento/module-page-builder/view/adminhtml/web/template/page-builder.html
+++ b/vendor/magento/module-page-builder/view/adminhtml/web/template/page-builder.html
@@ -31,6 +31,7 @@
         <with args="stage">
             <render></render>
         </with>
-        <iframe attr="id: 'render_frame_' + id" sandbox="allow-scripts allow-same-origin" style="position: absolute; width:0; height:0; border: none;"></iframe>
+        
+        <iframe attr="id: 'render_frame_' + id" sandbox="allow-scripts" style="position: absolute; width:0; height:0; border: none;"></iframe>
     </if>
 </div>`

**Comment by hostep** [comment]
Adding `allow-same-origin` is considered a security issue according to the comment in the patch mentioned above. So I wouldn't do that.

Based on the patch above (thanks @chrisoliver90!), here are commits that fix this in Magento OS modules:
- https://github.com/magento/magento2-page-builder/commit/70634ffe92c416e70c5c477f95619850b11f1a9d
- https://github.com/magento/magento2/commit/ab5947b73e08fef9f1b02d1c31b5ea882410f685

**Comment by medigeek** [comment]
@hostep I don't understand, was this released with 2.4.7 / 2.4.7-p2 ? Is this patch available somewhere? Is this added to the release notes? AFAICS this affects 2.4.7-p2.

**Comment by medigeek** [comment]
OK so I tried out and noticed that the latest versions of backend and page builder modules don't have this code included.

Having another go at this, these are the composer-compatible patches:

[fix-backend-error-page-builder-when-editing-cms-block.patch](https://github.com/user-attachments/files/16740623/fix-backend-error-page-builder-when-editing-cms-block.patch)

[fix-error-page-builder-when-editing-cms-block.patch](https://github.com/user-attachments/files/16740624/fix-error-page-builder-when-editing-cms-block.patch)

**Comment by engcom-Bravo** [comment]
@hostep @chrisoliver90 Thanks for your Contribution!!

As per this comment https://github.com/magento/magento2/issues/39076#issuecomment-2306918383 we are closing this issue.

Thanks.

**Comment by medigeek** [comment]
@engcom-Bravo was this issue documented in 2.4.7 and 2.4.7-p2 release notes? When can we expect this bugfix issue to be released? Thank you.

**Comment by sunit9** [comment]
Is any one has tried this solution into Magento CE 2.4.7-p2 ? I tired but not working.

**Comment by medigeek** [comment]
@sunit9 it should, as far as I can see the change in code removed the `allow-same-origin` but didn't include the workaround code in the patch. 
After the patch you have to do a full redeploy. 
If it doesn't work, check your full javascript logs for maybe some other errors to solve the issue.
Also, if you're using cloudflare, try without the cloudflare proxy.

**Comment by daniel-zaretsky1** [comment]
For anyone here with 2.4.6 and experiencing the `[ERROR] Page Builder was rendering for 5 seconds without releasing locks.  `
This patch fixed it for us: 
https://github.com/magento/quality-patches/blob/84479c0f3c57a32d7a903e9436f0e192e100e1c2/patches/os/ACSD-52041_2.4.6.patch

**Comment by hostep** [comment]
[Magento's quality patches](https://github.com/magento/quality-patches/) 1.1.50 got released today and also contains patches for this problem for various versions of Magento: https://github.com/magento/quality-patches/compare/e8ad81921a...d2e5320f05#diff-5c6d9dd586fe0c5875aa848aad7a1d0729ff28ff44d8bb8d3bec92dfa02f3207R12282-R12311

**Comment by sunit9** [comment]
@medigeek I got following error in console after apply 

[fix-backend-error-page-builder-when-editing-cms-block.patch](https://github.com/user-attachments/files/16740623/fix-backend-error-page-builder-when-editing-cms-block.patch)

[fix-error-page-builder-when-editing-cms-block.patch](https://github.com/user-attachments/files/16740624/fix-error-page-builder-when-editing-cms-block.patch)

[2024-08-28 02:37:57] [ERROR] Page Builder was rendering for 5 seconds without releasing locks.

![image](https://github.com/user-attachments/assets/3099c461-f7e4-48ac-9412-3703d556776d)

I am working on my local system. So no cloudflare proxy

**Comment by amiyoghoshstar** [comment]
Same happneing with me! Is it related to chrome browser?

**Comment by anjali25021995** [comment]
> Contributor

I treid this patch but it is not working magento 2.4.7-p2, please fix this bug as soon as possible

**Comment by daniel-zaretsky1** [comment]
After applying the new patch released ACSD-59514 we are still experiencing this issue in all browsers tested (Chrome/Firefox/Safari). Recompiled we experience that this issue sporadically resurfaces when saving. Sometimes the saves are successful in Firefox, other times we get the `[ERROR] Page Builder was rendering for 5 seconds without releasing locks`. In almost all cases using chrome it fails. Before our workaround was to just use Firefox to update CMS Blocks/Pages but not anymore since we're experiencing this in it as well now.

**Comment by anjali25021995** [comment]
@daniel-zaretsky1 Yes, we are not getting the every time the issue was coming sporadically resurfaces when saving, sometimes the save successful(Chrome/Firefox/), and other times we get the [ERROR] Page Builder was rendering for 5 seconds without releasing locks, permanent fix, can you let  us know we are using open source Magento 2,4,7-p2

**Comment by medigeek** [comment]
try adding something similar to your app/etc/env.php
you probably need to adapt your `env.php` file per use case (it contains a php array, consult with your developer).
do a full re-deploy/re-compile.

it might be regarded as a security issue, but it will probably make your site work.

```
    'system' => [
        'default' => [
            'csp' => [
               'mode' => [
                  'storefront_checkout_index_index' => [
                     'report_only' => '1'
                  ],
                  'admin_sales_order_create_index' => [
                    'report_only' => '1'
                  ]
               ],
               'policies' => [
                  'storefront_checkout_index_index' => [
                    'scripts' => [
                        'inline' => '1'
                    ]
                  ],
                  'admin_sales_order_create_index' => [
                    'scripts' => [
                        'inline' => '1'
                    ]
                  ]
               ]
            ]
        ]
    ]
```

**Comment by anjali25021995** [comment]
Magento 2 version 2.4.7-p2(CE), “Page Builder was rendering for 5 seconds without releasing locks” error. we applied the patch but still got the error ACSD-59582_2.4.7.patch, anyone please provide the solution

**Comment by medigeek** [comment]
did you guys do a `composer update` ?
which version of `composer info -a magento/module-page-builder` are you using?
you ought to be using v2.2.5-p2 (marked with an asterisk next to it)

**Comment by anjali25021995** [comment]
Yes, we are using composer update, our page builder version is  v2.2.5-p2 with magento2.4.7-p2, this page builder version correctly still we are getting the issue "“Page Builder was rendering for 5 seconds without releasing locks”

**Comment by medigeek** [comment]
@anjali25021995 try reproduce the issue in plain vanilla magento (default theme and default clean magento installation) if possible.

**Comment by sunit9** [comment]
@medigeek Not working in vanilla installation. I tried env.php solution also.

@engcom-Bravo  Please re-open this issue.

**Comment by anjali25021995** [comment]
Hi,

Anyone can help My account section->Address book by default default shipping address and default billing address checkbox not showing, can anyone help me this issue

**Comment by chrisoliver90** [comment]
> ACSD-59582_2.4.7

The patch I provided (ACSD-59582_2.4.7) was sent to us directly from Adobe Commerce. 
I don't believe a quality patch is in place yet but they replicated the issue and sent us that patch which worked.

**Comment by jmonrove** [comment]
This is still an issue

**Comment by medigeek** [comment]
@sunit9 try the patch from chris in comment above: https://github.com/magento/magento2/issues/39076#issuecomment-2306661703

**Comment by sunit9** [comment]
@medigeek this patch looks like. So it may not work.

https://github.com/magento/magento2-page-builder/commit/70634ffe92c416e70c5c477f95619850b11f1a9d
https://github.com/magento/magento2/commit/ab5947b73e08fef9f1b02d1c31b5ea882410f685

This may work in Adobe commerce but not working in Community Edition. I think need to re-open issue for CE.

**Comment by medigeek** [comment]
@sunit9 you could reopen the issue or open a new issue at page builder: https://github.com/magento/magento2-page-builder/issues
also mention this issue and this discussion ( https://github.com/magento/magento2/issues/39076 )

**Comment by Watch3ruk** [comment]
issue is not resolved on MOS . 
Could this be reopened please

**Comment by medigeek** [comment]
well in our case it returned, the solution was to add the nginx configuration headers as mentioned above...

```
#Fix CORS and magento page builder issue
#https://github.com/magento/magento2/issues/39076
#https://magento.stackexchange.com/a/170726
    add_header 'Access-Control-Allow-Origin' '*' always;
#add_header 'Access-Control-Allow-Credentials' 'true';
    add_header 'Access-Control-Allow-Methods' 'GET, POST, PUT, DELETE, OPTIONS' always;
    add_header 'Access-Control-Allow-Headers' 'DNT,Authorization,X-CustomHeader,Keep-Alive,User-Agent,Origin,Referer,X-HTTP-Method-Override,X-Accept-Charset,X-Accept,Accept,Access-Control-Request-Method,Access-Control-Request-Headers,X-Requested-With,If-Modified-Since,Cache-Control,Content-Type' always;
```

secure or not, it works. until magento open source provides a proper fix for this.
We don't all have access to ACSD patches unfortunately.

**Comment by hostep** [comment]
To [repeat myself](https://github.com/magento/magento2/issues/39076#issuecomment-2314440918), there are patches included for this in the magento quality patches repository, you can find them in the [os](https://github.com/magento/quality-patches/tree/master/patches/os) and [commerce](https://github.com/magento/quality-patches/tree/master/patches/commerce) directories.

They are [defined](https://github.com/magento/quality-patches/compare/e8ad81921a...d2e5320f05#diff-5c6d9dd586fe0c5875aa848aad7a1d0729ff28ff44d8bb8d3bec92dfa02f3207R12282-R12311) like this for various versions of Magento and should be applied to 2 different magento packages:
```json
    "ACSD-59514": {
        "categories": [
            "Admin"
        ],
        "title": "Fixes the issue where forms in the Admin area with Page Builder throw the error \"Page Builder was rendering for 5 seconds without releasing locks.\" in the browser console after submitting the form, and changes can't be saved.",
        "packages": {
            "magento/magento2-base": {
                ">=2.4.4 <2.4.4-p9 || >=2.4.5 <2.4.5-p8 || >=2.4.6 <2.4.6-p6": {
                    "file": "os/ACSD-59514_2.4.4-p8.patch"
                },
                ">=2.4.4-p9 <2.4.5 || >=2.4.5-p8 <2.4.6 || >=2.4.6-p6 <2.4.7": {
                    "file": "os/ACSD-60169_2.4.6-p6.patch"
                },
                ">=2.4.7 <2.4.7-p2": {
                    "file": "os/ACSD-59582_2.4.7.patch"
                }
            },
            "magento/module-page-builder": {
                ">=2.2.2 <2.2.2-p9 || >=2.2.3 <2.2.3-p8 || >=2.2.4 <2.2.4-p6": {
                    "file": "commerce/ACSD-59514_2.2.2-p7.patch"
                },
                ">=2.2.2-p9 <2.2.3 || >=2.2.3-p8 <2.2.4 || >=2.2.4-p6 <2.2.5": {
                    "file": "commerce/ACSD-60169_2.2.4-p6.patch"
                },
                ">=2.2.5 <2.2.5-p2": {
                    "file": "commerce/ACSD-59582_2.2.5.patch"
                }
            }
        }
    },
```

**Comment by medigeek** [comment]
@hostep I admire the commitment but unfortunately for some reason [ACSD-59582_2.4.7.patch](https://github.com/magento/quality-patches/blob/master/patches/os/ACSD-59582_2.4.7.patch) on 2.4.7-p2 doesn't work when applied (i.e. successfully applied but doesn't fix the issue). The only fix that would work was the nginx in my case. mileage may vary obviously.

**Comment by ElSamhaa** [comment]
Hi, installing from the quality patches didn't resolve it for me. `requirejs-config.js` is not returned (404) even though the file is generated.

**Comment by Watch3ruk** [comment]
fixed this for me : 
check that around line 78 in .../vendor/magento/module-require-js/Block/Html/Head/Config.php
    protected function _prepareLayout()
    {
        $after = $this->getInsertAfter() ?: RequireJsConfig::REQUIRE_JS_FILE_NAME;
not 
$after = RequireJsConfig::REQUIRE_JS_FILE_NAME;

**Comment by aamoronatti** [comment]
I applied all the patches and It wasn't working for me. I purged cloudflare's cache and the page started saving correctly. Maybe this might help someone.

**Comment by vivekmehta17** [comment]
Hi All,

Tried solution for above issue on Magento2.4.7-p2 working fine and page builder issue not coming now.

create app/code/Vendor/Module/view/adminhtml/requirejs-config.js
var config = {
    map: {
        '*': {
            'Magento_PageBuilder/template/page-builder.html': 'Vendor_Module/template/page-builder.html'

        },
    }
};

app/code/Vendor/Moduleview/adminhtml/web/template/

<div class="pagebuilder-stage-wrapper"
     css="Object.assign(
        {'stage-full-screen': isFullScreen, 'stage-content-snapshot': isSnapshot, 'transition-out': isSnapshotTransition, 'stage-is-active': (!isSnapshot() || isFullScreen)},
        viewportClasses)"
     ko-style="wrapperStyles">
    <div class="admin__field pagebuilder-header"
         css="visible: panel.isVisible(), 'transition-out': isSnapshotTransition()"
         if="!isSnapshot()">
        <span class="viewport-buttons">
              <render args="viewportTemplate"></render>
        </span>
        <span class="template-buttons">
            <button type="button" translate="'Apply Template'" if="isAllowedTemplateApply" click="toggleTemplateManger"></button>
            <button type="button" translate="'Save as Template'" if="isAllowedTemplateSave" click="saveAsTemplate"></button>
            <i click="toggleFullScreen.bind($data, {animate:true})"
               css="'icon-pagebuilder-fullscreen': !isFullScreen(), 'icon-pagebuilder-fullscreen-exit': isFullScreen()"
               attr="{title: isFullScreen() ? $t('Close Full Screen') : $t('Open in Full Screen')}">
            </i>
        </span>
    </div>
    <with args="panel">
        <render></render>
    </with>
    <if args="isStageReady">
        <with args="stage">
            <render></render>
        </with>
        
        <iframe attr="id: 'render_frame_' + id" sandbox="allow-scripts allow-same-origin" style="position: absolute; width:0; height:0; border: none;"></iframe>
    </if>
</div>

**Comment by smartexcan** [comment]
I believe I've found the root of the CORS issue described in https://github.com/magento/magento2/issues/39076#issuecomment-2303063100

Earlier this year, changes were made to the Csp module to add subresource integrity to js files.

While originally intended to only be added for checkout and order payment pages, currently it is applied to the 'requirejs-config.js' file on every page in storefront and adminhtml.

We're currently still using the 2.4.6 release line, and these changes look to have been added in 2.4.6-p6.

I created a small module to fix the issue only on the pagebuilder_stage_render page.
https://github.com/smartexcan/magento2-page-builder-require-js-fix

**Comment by hostep** [comment]
Thanks for the hint @smartexcan, I'll link to https://github.com/magento/magento2/issues/39250 which is now very closely related to this issue I believe.

**Comment by chalov-anton** [comment]
Still an issue on 2.4.7-p3
I'm unable to save a block with HTML Element added via Page Builder.

![Image](https://github.com/user-attachments/assets/17f022c7-a69b-49dd-b4c9-615141a49535)

**Comment by ssx** [comment]
> Still an issue on 2.4.7-p3 I'm unable to save a block with HTML Element added via Page Builder.
> 
> ![Image](https://github.com/user-attachments/assets/17f022c7-a69b-49dd-b4c9-615141a49535)

you could possibly try: https://github.com/magento/magento2-page-builder/issues/866#issuecomment-2669836572

**Comment by Kishorchhugani** [comment]
Still an issue on 2.4.7-p4
I'm unable to save a block with HTML Element added via Page Builder.
sometimes products unable to save 
cms pages unable to save 
same error : Page Builder was rendering for 5 seconds without releasing locks.

**Comment by gshep86** [comment]
We are on 2.4.7-p8 and are still seeing this issue even after applying the patch

**Comment by resplin** [comment]
We solved this on 2.4.6-p8 by applying patch **ACSD-52041** (os/ACSD-52041_2.4.6.patch) although it is (incorrectly) marked as deprecated.

**Comment by FY0u11** [comment]
Still an issue, but this issue is closed, why? 2.4.7-p2 - what patches should I apply?

**Comment by FY0u11** [comment]
https://github.com/magento/magento2/issues/39076#issuecomment-2306918383 This doesn't work on 2.4.7-p2

**Comment by hostep** [comment]
Does the following (yet another) quality patch help [ACP2E-3838_2.4.7-p3.patch](https://github.com/magento/quality-patches/blob/6f2d758b05091efcbf9f188a2163a2bba4ed8c60/patches/os/ACP2E-3838_2.4.7-p3.patch)? (mind that it has a dependency on patch [ACSD-63244](https://github.com/magento/quality-patches/blob/6f2d758b05091efcbf9f188a2163a2bba4ed8c60/support-patches.json#L13377-L13401)).
Its definition lives [here](https://github.com/magento/quality-patches/blob/6f2d758b05091efcbf9f188a2163a2bba4ed8c60/support-patches.json#L14790-L14806)


I'm only guessing here with all my comments, I haven't witnessed this problem myself yet in any of the shops we maintain...
